lib/Common/Memory/arm64/XDataAllocator.cpp

```diff
@@ -3,87 +3,58 @@
 // Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
 //-------------------------------------------------------------------------------------------------------
 // This one works only for ARM64
-#include "stdafx.h"
+#include "CommonMemoryPch.h"
 #if !defined(_M_ARM64)
 CompileAssert(false)
 #endif
 
-XDataAllocator::XDataAllocator(BYTE* address, uint size)
+#include "XDataAllocator.h"
+#include "Core/DelayLoadLibrary.h"
+
+XDataAllocator::XDataAllocator(BYTE* address, uint size, HANDLE processHandle) :
+    processHandle(processHandle)
 {
-    Assert(size == 0);
+    __debugbreak();
 }
 
 
 void XDataAllocator::Delete()
 {
-    HeapDelete(this);
+    __debugbreak();
 }
 
 bool XDataAllocator::Initialize(void* segmentStart, void* segmentEnd)
 {
+    __debugbreak();
     return true;
 }
 
 bool XDataAllocator::Alloc(ULONG_PTR functionStart, DWORD functionSize, ushort pdataCount, ushort xdataSize, SecondaryAllocation* allocation)
 {
-    XDataAllocation* xdata = static_cast<XDataAllocation*>(allocation);
-    Assert(pdataCount > 0);
-    Assert(xdataSize >= 0);
-    Assert(xdata);
-    bool success = true;
-
-    DWORD size = GetAllocSize(pdataCount, xdataSize);
-    BYTE* alloc = HeapNewNoThrowArray(BYTE, size);
-    if(alloc != null)
-    {
-        xdata->address = alloc;
-        xdata->xdataSize = xdataSize;
-        xdata->pdataCount = pdataCount;
-    }
-    return success;
+    __debugbreak();
+    return false;
 }
 
-void XDataAllocator::Register(XDataAllocation& allocation, ULONG_PTR functionStart, DWORD functionSize)
+
+void XDataAllocator::Release(const SecondaryAllocation& allocation)
 {
-    RUNTIME_FUNCTION* pdataArray = allocation.GetPdataArray();
-    for(ushort i = 0; i < allocation.pdataCount; i++)
-    {
-        RUNTIME_FUNCTION* pdata = pdataArray + i;
-        Assert(pdata->UnwindData != 0);
-        Assert(pdata->BeginAddress != 0);
-        pdata->BeginAddress = (DWORD)((ULONG_PTR)pdata->BeginAddress - functionStart);
-        if(pdata->Flag != 1) // if it is not packed unwind data
-        {
-            pdata->UnwindData = (DWORD)((ULONG_PTR)pdata->UnwindData - functionStart);
-        }
-    }
-    Assert(allocation.functionTable == null);
+    __debugbreak();
```

Comment:
Why are we replacing all the implementations with a no-op/break depending on configuration?